ports/esp32/mpthreadport.c

```diff
@@ -161,7 +161,7 @@ mp_uint_t mp_thread_create_ex(void *(*entry)(void *), void *arg, size_t *stack_s
     thread = th;
 
     // adjust the stack_size to provide room to recover from hitting the limit
-    *stack_size -= 1024;
+    *stack_size -= MP_TASK_STACK_LIMIT_MARGIN;
```

Comment:
This code has a pre-existing race condition, because stack_size points to inside the thread 'arg' structure (and the thread is already created. If a context switch happens before here then it will set the limit to the full task stack size.

It still failed for me on IDF V5.2.2 until I moved the adjustment before creating the RTOS thread. However I think I found a simpler way to handle it, will submit PR after lunch.

(As an aside, I thought I saw the stress_recurse test failing intermittently on esp32 - maybe this is why.)

Reply:
Great find!  I'll wait for your fix then.

Reply:
(Alternative now at https://github.com/micropython/micropython/pull/15605 )